ui/jstests/test_manage_taxonomies.jsx

```diff
@@ -35,39 +35,278 @@ define(['QUnit', 'jquery', 'setup_manage_taxonomies', 'reactaddons',
       }
     ]
   };
-  function reportError(msg) {
-    if (msg) {
-      console.error(msg);
-    }
+  /**
+   * Common assertions for Add terms, use in term and taxonomy
+   * components tests
+   * @param {Object} assert - For assertions
+   * @param {Object} component - Root DOM object in react class
+   * @return {Object} devItems - DOM object of class 'panel-body'
+   * */
+  function  assertAddTermCommon(assert, component) {
+    // check term title
+    var listOfLinks = React.addons.TestUtils.
+      scryRenderedDOMComponentsWithTag(
+        component,
+        'a'
+    );
+    assert.equal(listOfLinks.length, 2);
+    var linkHeader = listOfLinks[0].getDOMNode();
+    assert.equal(linkHeader.innerHTML, vocabulary.name);
+    //test items
+    var devItems = React.addons.TestUtils.
+      findRenderedDOMComponentWithClass(
+        component,
+        'panel-body'
+    );
+    var itemList = React.addons.TestUtils.
+      scryRenderedDOMComponentsWithTag(
+        devItems,
+        'li'
+    );
+    assert.equal(itemList.length, 2);
+    return devItems;
+  }
+  /**
+   * Common assertions for Add Vocabulary, use in term and taxonomy
+   * components tests
+   * @param {Object} assert - For assertions
+   * @param {Object} component - Root DOM object in react class
+   * @return {Object} devItems - DOM object of class 'form-horizontal'
+   * */
+  function assertAddVocabularyCommon(assert, component) {
+    var formNode = React.addons.TestUtils.
+      findRenderedDOMComponentWithClass(
+        component,
+        'form-horizontal'
+      );
+    assert.ok(formNode);
+    //test form submission
+    var inputNodes = React.addons.TestUtils.
+      scryRenderedDOMComponentsWithTag(
+        formNode,
+        'input'
+      );
+    assert.equal(inputNodes.length, 12);
+    var inputVocabularyName = inputNodes[0];
+    var inputVocabularyDesc =  inputNodes[1];
+    var radioManaged =  inputNodes[10];
+    var radioTagStyle = inputNodes[11];
+
+    React.addons.TestUtils.Simulate.change(
+      inputVocabularyName, {target: {value: "TestA"}}
+    );
+    React.addons.TestUtils.Simulate.change(
+      inputVocabularyDesc, {target: {value: "TestA"}}
+    );
+    radioManaged.checked = true;
+    React.addons.TestUtils.Simulate.change(
+      radioManaged,
+      {target: {value: 'f'}}
+    );
+    radioTagStyle.checked = false;
+    var done = assert.async();
+    component.forceUpdate(function() {
+      assert.equal(component.state.name, "TestA");
```

Comment:
This can cause a race condition in the tests because `forceUpdate` will run this piece of code asynchronously but `Simulate.submit` below will trigger an AJAX request which also runs asynchronously. This should be reorganized to put `Simulate.submit` inside the `forceUpdate` callback to ensure they happen in that order then and have `done` execute after the ajax finishes.

It would probably be easiest to inline this function since it's only used in one place. Otherwise you would need to create and return a promise to ensure proper ordering of these callbacks.